### ISSUE TITLE
Jekyll Generators

### ISSUE THREAD

**Issue by aylalab** [aylalab/ecommerce-pattern-library#3]
Hi @thomasjbradley,

Sorry for bugging you again, I'm just very confused about how to go about this section with my code. I already have the product cards individually styles and clickable, so I've been trying to figure out how to adapt the code from this assignment to my website without messing up what I've already done. Am I able to set up the products list page with the code I already have, or will I have to re-write it to incorporate the Jekyll elements like in this assignment?

Thanks!

**Comment by thomasjbradley** [comment]
I don’t see any code related to loops & generating product cards in your repo. Did you commit & push?

**Comment by aylalab** [comment]
Yes I did, it might be because I've placed them in the completely wrong place. I'm confused about how to add this code into my already styled product cards without disrupting the styling I've already done (they're in _patterns > cards > products.html). Would you be able to give me some guidance on this? I'm very lost at the moment and am not sure how to proceed.

Thank you!

@thomasjbradley

**Comment by thomasjbradley** [comment]
1. The `products.html` pattern should contain only 1 single product card—not all of them.
2. The product title in there should be replaced with an include placeholder: `{{include.data.name}}` (matching what’s in your `.md` files)
3. Do the same for the `<img src="">`
4. The `<a href="">` should be changed to a placeholder too: `<a href="{{include.data.url}}">`

---

Then you’ll need the generator code within your `products.html` file:
https://learn-the-web.algonquindesign.ca/courses/web-dev-4/jekyll-generators/#step-3

**Comment by aylalab** [comment]
Okay so I've made the changes and I think the code is almost there, but I'm not seeing the cards on the website? Also I think I've lost the css styling I've done for each of the three cards, how would I go about adding those classes again now that there's only one card in the html?

@thomasjbradley

**Comment by thomasjbradley** [comment]
This code should be in the `products.html` page, in the root folder:
https://github.com/aylalab/ecommerce-pattern-library/blob/master/_patterns/cards/products.html#L8-L10

**Comment by aylalab** [comment]
Yay I finally got them to show up! I only have two more questions about this for you;

1 - This might be a dumb question, but how do I fix the weird spacing around the cards?  I want them to be centre and to fill the page evenly, and for each card to be a bit wider than they currently are.

2 - I had assigned each different type of coffee a different colour. How do I target each individual type to display as a different colour.

Thank you so much for your help, I really appreciate it :)

@thomasjbradley

**Comment by thomasjbradley** [comment]
The spacing, I think, is related to the `.about` class, which has width & height on it.

**Comment by thomasjbradley** [comment]
Colours: that’s in the cards too?
If so, that info will need to be inside the `.md` files—to define what colour they’re supposed to be.

**Comment by aylalab** [comment]
I tried removing the width and height from the .about class and it did not affect the product cards. I also tried adding width and height to the cards themselves and it still did not work. Could it be the grid interfering in some way?

Also, how do I go about targeting each card inside the .md file to change their colour?

Thank you!

@thomasjbradley
